lesson_04_hw_04/views_database.py

```diff
@@ -0,0 +1,88 @@
+"""Implement the following view functions.
+"""
+
+import sqlite3
+
+from flask import Flask
+from webargs import fields, validate
+from webargs.flaskparser import use_kwargs
+
+
+app = Flask(__name__)
+
+
+def execute_query(query, args=()):
+    with sqlite3.connect('chinook.db') as connection:
+        cursor = connection.cursor()
+        cursor.execute(query, args)
+        connection.commit()
+        records = cursor.fetchall()
+    return records
+
+
+def format_records(records: list) -> str:
+    return '<br>'.join(str(record) for record in records)
+
+
+@app.route("/order-price")
+@use_kwargs(
+    {
+        'country': fields.Str(
+            load_default=None)
+    },
+    location='query'
+)
+def order_price(country):
+    """ Function calculate sales by country on page / if no country show all sales by all counties.
+    """
+    if country:
+        query = "SELECT SUM(invoice_items.UnitPrice * invoice_items.Quantity) AS Sales, invoices.BillingCountry" \
+            " FROM invoice_items JOIN invoices ON  invoice_items.InvoiceId==invoices.InvoiceId " \
+            f"GROUP BY invoices.BillingCountry HAVING invoices.BillingCountry=='{country}'"
+    else:
+        query = "SELECT SUM(invoice_items.UnitPrice * invoice_items.Quantity) AS Sales, invoices.BillingCountry" \
+            " FROM invoice_items JOIN invoices ON  invoice_items.InvoiceId==invoices.InvoiceId " \
+            "GROUP BY invoices.BillingCountry"
+
+    records = execute_query(query)
+    return format_records(records)
+
+
+@app.route("/track-info")
+@use_kwargs(
+    {
+        'id': fields.Int(
+            load_default=None,
+            validate=[validate.Range(min=1, max=500)]
+        )
+    },
+    location='query'
+)
+def get_all_info_about_track(id):
+    """Function gives information about track.
+    """
+    query = "SELECT tracks.TrackId, tracks.Name, tracks.Composer, albums.Title," \
+            " artists.Name, genres.Name, media_types.Name" \
+            " FROM tracks JOIN albums ON tracks.AlbumId = albums.AlbumId" \
+            " JOIN artists ON albums.ArtistId = artists.ArtistId" \
+            " JOIN genres ON tracks.GenreId = genres.GenreId" \
+            " JOIN media_types ON tracks.MediaTypeId = media_types.MediaTypeId" \
+            f" GROUP BY tracks.TrackId HAVING tracks.TrackId=='{id}'"
+
+    records = execute_query(query)
```

Comment:
Add handling when the id from the user is not located in the database. Show the message that the track id is missing.